src/rebalancers/uniswap-v3/UniswapV3Rebalancer.sol

```
@@ -215,10 +222,19 @@ contract UniswapV3Rebalancer is IActionBase {
             // Rebalance the position so that the maximum liquidity can be added for new ticks.
             // The Pool must still be balanced after the swap.
             (bool zeroToOne, uint256 amountIn) = getSwapParameters(position, amount0, amount1, initiatorFee);
+
             // Get initiator fee amount and deduct from amountIn.
             uint256 feeAmount = amountIn.mulDivDown(initiatorFee, 1e18);
             amountIn -= feeAmount;
-            if (_swap(position, zeroToOne, amountIn)) revert UnbalancedPool();
+
+            if (swapData.length > 0) {
+                // Perform arbitrary swap
+                // TODO : if not a success go via uniV3 swap ?
```

Not needed, just let it fail no need to make it even more complex